app/controllers/katello/api/v2/host_tracer_controller.rb

```diff
@@ -16,6 +16,45 @@ def index
       respond_for_index(:collection => collection)
     end
 
+    api :PUT, "/traces/resolve", N_("Resolve Traces")
+    param :trace_ids, Array, :required => true, :desc => N_("Array of Trace IDs")
+    def resolve
+      traces = Katello::HostTracer.resolvable.where(id: params[:trace_ids])
+
+      traces.each do |trace|
+        if trace.helper.include?('reboot')
+          trace.helper = 'reboot'
+        end
+      end
+
+      traces_by_host_id = traces.group_by(&:host_id)
+      traces_by_helper = traces.group_by(&:helper)
+
+      composers = []
+
+      if traces_by_host_id.size < traces_by_helper.size
```

Comment:
None of the tests actually test the code within this `if` statement - just the else. I found it by stepping through the code. I think if we update one of the tests to add an additional trace with a different helper for one of the hosts we already have, that will do the trick